### ISSUE TITLE
REGRESSION: Performance

### ISSUE THREAD

**Issue by boyter** [boyter/scc#472]
**Describe the bug**

The latest release which includes the new git logic pulled from gocodewalker has made scc much slower than is used to be.

**To Reproduce**

Compare version before and after the change.

**Expected behavior**

Would be nice to have the speed back.

**Comment by boyter** [comment]
First tweak by modifying gocodewalker to walk in parallel and cache some calls. Still more work needed to close the gap, as on a 32 core machine scc counts the linux kernel in ~1 second while tokei is ~700ms. 

![image](https://github.com/boyter/scc/assets/612151/886ddbb7-7c25-475d-b777-076487e0ba05)

**Comment by boyter** [comment]
Going to create release with part one which gets most of the performance back.

**Comment by boyter** [comment]
Gap mostly closed with the latest release.
